x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/settings_flyout/schedule/create_flyout/index.tsx

```diff
@@ -32,16 +41,75 @@ export const CreateFlyout: React.FC<Props> = React.memo(({ onClose }) => {
     prefix: 'attackDiscoveryScheduleCreateFlyoutTitle',
   });
 
+  const {
+    services: { uiSettings },
+  } = useKibana();
+
+  const { alertsIndexPattern, http } = useAssistantContext();
+  const { data: aiConnectors, isLoading: isLoadingConnectors } = useLoadConnectors({
+    http,
+  });
+  const [isCreatingSchedule, setIsCreatingSchedule] = useState(false);
+
+  const { sourcererDataView } = useSourcererDataView();
+
+  const { mutateAsync: createAttackDiscoverySchedule } = useCreateAttackDiscoverySchedule();
```

Comment:
Consider returning the loading state from this hook to avoid local state

Reply:
done in https://github.com/elastic/kibana/pull/217917/commits/38585fbef19ecfdf1ef2764d67edeead4f00329e